macro/run_simScript.py

```diff
@@ -291,6 +299,23 @@
 # P8gen.SetMom(500.*u.GeV)
 # P8gen.SetId(-211)
  primGen.AddGenerator(P8gen)
+ 
+if simEngine == "ALPACA":
+  target     = ship_geo.target
+  z_middle   = target.z0
+  DZ         = target.length/2
+  startZ     =   z_middle - DZ
+  endZ       =   z_middle + DZ
+  SmearBeam  = 1*u.cm
+  print(SmearBeam,"run")
+  Lmin       = ((ship_geo.Chamber1.z - ship_geo.chambers.Tub1length) - ship_geo.target.z0)/100.
+  Lmax       = (ship_geo.TrackStation1.z - ship_geo.target.z0)/100.
+  inputFile  = makeALPACAEvents.runEvents(options.theMass,options.theDPepsilon,options.nEvents,Lmin,Lmax,startZ,endZ,SmearBeam)
```

Comment:
Maybe you could print a log message here to tell the user that ALPACA is generating the input file?

Reply:
thanks it is done

Reply:
From the message in 303 I don't think it's clear that an input file is generated. I would add a message before and after `makeALPACAEvents.runEvents`. Ideally also one after opening the file, so that in future it's easier to debug if e.g. the file can't be opened, or something went wrong while generating the file.

Reply:
[this line](https://github.com/ShipSoft/FairShip/blob/f4baf2fbe05c9ad9f5d87f6df7906f1562957893/shipgen/ALPACAGenerator.cxx#L25) tells it is generated and opening.
[This one](https://github.com/ShipSoft/FairShip/blob/f4baf2fbe05c9ad9f5d87f6df7906f1562957893/shipgen/ALPACAGenerator.cxx#L28) gives an error message if the input is empty.
[In here](https://github.com/ShipSoft/FairShip/blob/f4baf2fbe05c9ad9f5d87f6df7906f1562957893/shipgen/ALPACAGenerator.cxx#L72), it tells us the nth event is reading.
For the end message, I will fix [the condition](https://github.com/ShipSoft/FairShip/pull/332#discussion_r397204266).
Also for the ALPACA generation, It has it's own message e.g. as Pythia.

Reply:
I saw the messages in the generator (and they are very informative, but did not see any in the makeALPACAEvents script.

If ALPACA has its own messages, i would suggest having a message before entering it and after leaving it. My main worry is that it might be hard to debug right now if e.g. the ALPACA file isn't written properly (maybe due to some AFS issue), but ALPACA claims it was successful. Then you might not have any indication of an error until the generator tries to open the file.

Reply:
I added printouts into [run_simScripy](https://github.com/ShipSoft/FairShip/blob/6dddb88070d7416db8ebbe02081dd940a3a75d9a/macro/run_simScript.py), [makeALPACAEvents](https://github.com/ShipSoft/FairShip/blob/6dddb88070d7416db8ebbe02081dd940a3a75d9a/python/makeALPACAEvents.py) and [ALPACAGenerator](https://github.com/ShipSoft/FairShip/blob/6dddb88070d7416db8ebbe02081dd940a3a75d9a/shipgen/ALPACAGenerator.cxx).
Can you have a look at the content of messages?

Reply:
Looks good to me! Thanks!

Reply:
Thanks!